src/coap_io.c

```diff
@@ -258,6 +258,11 @@ coap_network_send(struct coap_context_t *context UNUSED_PARAM,
 #ifndef WITH_CONTIKI
   /* a buffer large enough to hold all protocol address types */
   char buf[CMSG_LEN(sizeof(struct sockaddr_storage))];
+  (void)datalen;
```

Comment:
You're right - sorry about that. It was a result of a merge issue I missed. I moved this line and the next to the "else" part of "#ifndef WITH_CONTIKI".